op-challenger/fault/responder.go

```diff
@@ -0,0 +1,163 @@
+package fault
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// faultResponder implements the [Responder] interface to send onchain transactions.
+type faultResponder struct {
+	log log.Logger
+
+	shutdownCtx       context.Context
+	cancelShutdownCtx context.CancelFunc
+
+	// Transaction manager and configuration.
+	txMgr                  txmgr.TxManager
+	MaxPendingTransactions uint64
+
+	// Fault Dispute Game contract.
+	fdgAddr     *common.Address
+	fdgContract *bindings.FaultDisputeGameTransactor
+}
+
+// NewFaultResponder returns a new [faultResponder].
+func NewFaultResponder(logger log.Logger, fdgAddress *common.Address, fdgContract *bindings.FaultDisputeGameTransactor) faultResponder {
+	shutdownCtx, cancelShutdownCtx := context.WithCancel(context.Background())
+	return faultResponder{
+		shutdownCtx:       shutdownCtx,
+		cancelShutdownCtx: cancelShutdownCtx,
+		log:               logger,
+		fdgAddr:           fdgAddress,
+		fdgContract:       fdgContract,
+	}
+}
+
+// Quit shuts down the [faultResponder].
+func (r *faultResponder) Quit() {
+	r.log.Info("Shutting down fault responder")
+	r.cancelShutdownCtx()
+}
+
+// buildTxData builds the transaction data for the [faultResponder].
+func (r *faultResponder) buildTxData(ctx context.Context, response Claim) (TxData, error) {
+	// Get the parent index as a big integer.
+	parentIndex := response.Parent.ToGIndex()
+	bigParentIndex := big.NewInt(int64(parentIndex))
```

Comment:
This is the wrong index. I guess we need to get some more information from the contract.
The index we need to include on chain is the index of the claim inside the ClaimdData array.
https://github.com/ethereum-optimism/optimism/blob/f419bd1514ee8ad5115ea52c2c09698f59eb48a9/packages/contracts-bedrock/contracts/dispute/FaultDisputeGame.sol#L53